### PR TITLE
OTC-369: Add fetching bulk control numbers to ControlNumberService

### PR DESCRIPTION
**Pull request by malinowskikam**
This PR contains changes from: [OTC-367](https://github.com/openimis/policies_android_app_java/pull/87), [OTC-368](https://github.com/openimis/policies_android_app_java/pull/88)

Changes: 
- Added new functionality to ControlNumberService to request and store control numbers from REST API.
- Modified BulkControlNumbersActivity to make use of the new functionality.
- Modified BulkControlNumbersActivity to better show which elements are interactive.
- Modified BulkControlNumbersActivity to allow to select a product for which control numbers should be requested.

Additional changes:
- Fixed the issue with not being able to install multiple versions of the Policies app
- Updated ToRestApi with changes made for Claims app
- Changed SQLHandler.insertData method to not throw UserExcetion, which was always silenced.
- updated gradle to 4.1.2

[https://openimis.atlassian.net/browse/OTC-369](https://openimis.atlassian.net/browse/OTC-369)

preview:
otc-369-1
otc-369-2
otc-369-3

### CODE REVIEW

File: app/src/main/java/org/openimis/imispolicies/ToRestApi.java [hunk 1]
```diff
@@ -12,137 +14,116 @@
...
-        HttpResponse response = null;
-        try {
-            response = httpClient.execute(httpPost);
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        return response;
+    public ToRestApi() {
+        token = Global.getGlobal().getJWTToken();
+        uri = AppInformation.DomainInfo.getDomain() + "api/";
```
Comment (kmadej) [inline app/src/main/java/org/openimis/imispolicies/ToRestApi.java:33]: move api string constant

File: app/src/main/java/org/openimis/imispolicies/ToRestApi.java [hunk 2]
```diff
@@ -12,137 +14,116 @@
...
-        try {
-            StringEntity postingString = new StringEntity(object.toString());
-            httpPost.setEntity(postingString);
-            httpPost.setHeader("Content-type", "application/json");
-        } catch (UnsupportedEncodingException e) {
-            e.printStackTrace();
+        HttpGet httpGet = new HttpGet(uri + functionName);
+        httpGet.setHeader("Content-Type", "application/json");
+        httpGet.setHeader("accept", "application/json");
+        httpGet.setHeader("api-version", apiVersion);
```
Comment (kmadej) [inline app/src/main/java/org/openimis/imispolicies/ToRestApi.java:42]: please try to find constant in http library

**Review by dragos-dobre — APPROVED** [review]

File: app/src/main/java/org/openimis/imispolicies/ToRestApi.java [hunk 3]
```diff
@@ -12,137 +14,116 @@
...
+                StringEntity postingString = new StringEntity(object.toString());
+                httpPost.setEntity(postingString);
             }
+            HttpResponse response = httpClient.execute(httpPost);
+            if (addToken && response != null && response.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                token.clearToken();
+            }
+            int responseCode = response.getStatusLine().getStatusCode();
+            Log.i("HTTP_POST", uri + functionName + " - " + responseCode);
+            return response;
```
Comment (kmadej) [inline app/src/main/java/org/openimis/imispolicies/ToRestApi.java:83]: I see similarity in this code with the method above. Please try to find common code and move to the separate method

**Review by kmadej — CHANGES_REQUESTED** [review]

**Comment by malinowskikam** [comment]
@kmadej i applied the changes

**Review by kmadej — APPROVED** [review]
@malinowskikam good work, it's looks much better

**Comment by kmadej** [comment]
@dragos-dobre I see your approval, so I'm merging PR

**Pull request merged**